features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -28,25 +28,63 @@
 
 package org.opennms.features.deviceconfig.rest.impl;
 
+import java.time.Duration;
 import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Collectors;
-
+import javax.ws.rs.PathParam;
 import javax.ws.rs.core.Response;
 
+import com.google.common.base.Strings;
 import org.apache.commons.lang3.StringUtils;
+import org.opennms.core.criteria.Criteria;
 import org.opennms.core.criteria.CriteriaBuilder;
+import org.opennms.core.utils.InetAddressUtils;
+import org.opennms.features.deviceconfig.persistence.api.ConfigType;
 import org.opennms.features.deviceconfig.persistence.api.DeviceConfig;
 import org.opennms.features.deviceconfig.persistence.api.DeviceConfigDao;
 import org.opennms.features.deviceconfig.rest.api.DeviceConfigDTO;
 import org.opennms.features.deviceconfig.rest.api.DeviceConfigRestService;
+import org.opennms.netmgt.dao.api.MonitoredServiceDao;
+import org.opennms.netmgt.model.OnmsIpInterface;
+import org.opennms.netmgt.model.OnmsMetaData;
+import org.opennms.netmgt.model.OnmsMonitoredService;
+import org.opennms.netmgt.model.OnmsNode;
 import org.opennms.web.utils.ResponseUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class DefaultDeviceConfigRestService implements DeviceConfigRestService {
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultDeviceConfigRestService.class);
+    private static final String REQUISITION_CONTEXT = "requisition";
+    private static final String SCHEDULE_METADATA_KEY = "schedule";
+
+    private static final Set<String> VALID_ORDER_BY_KEYS = Set.of("lastupdated", "devicename", "createdtime", "ipaddress");
```

Comment:
Maybe you could use a map: orderBy -> (hql property) here. Then the check if an `orderBy` is supported and its translation would be easy to implement.